### ISSUE TITLE
Search by storage id lasts freacking long on my company account

### ISSUE THREAD

**Issue by Kotsiubynskyi** [microsoft/AzureStorageExplorer#7239]
### Preflight Checklist

- [X] I have installed the [latest version of Storage Explorer](https://github.com/Microsoft/AzureStorageExplorer/releases/latest).
- [X] I have checked existing resources, including the [troubleshooting guide](https://docs.microsoft.com/azure/storage/common/storage-explorer-troubleshooting) and the [release notes](https://github.com/microsoft/AzureStorageExplorer/releases).
- [X] I have searched for [similar issues](https://github.com/Microsoft/AzureStorageExplorer/issues).

### Storage Explorer Version

1.31.1

### Regression From

1.30

### Architecture

x64

### Storage Explorer Build Number

20230822.1

### Platform

Windows

### OS Version

10

### Bug Description

Search by storage id lasts freacking long on my company account
it was much better several updates go but now it's veeeerrry long 

### Steps to Reproduce

1. Start app
2. In filter check only search by storage
3. input storage id and press enter to start searching
4. go to kitchen to drink coffee for about 5 minutes

### Actual Experience

take too much time

**Comment by Kotsiubynskyi** [comment]
![image](https://github.com/microsoft/AzureStorageExplorer/assets/4141297/b33051a5-ce9f-48d6-b6d7-c30e2741ee45)

**Comment by Kotsiubynskyi** [comment]
![image](https://github.com/microsoft/AzureStorageExplorer/assets/4141297/88af3e76-f060-4b20-852f-d324eabe40b4)

**Comment by JasonYeMSFT** [comment]
@Kotsiubynskyi You can right click the subscription and select "search from here" to scope the search in that subscription and reduce the unnecessary scanning. How many storage accounts do you have? Does it eventually give you the results or does it seem to hang forever?

**Comment by Kotsiubynskyi** [comment]
I have 10+ subscriptions

Thanks, your advice helped but not much. The search under one subscription takes now 3+ minutes instead of 5 for all subscriptions. I still do not think it's acceptable assuming that on Azure Portal the storage can be found within 2 seconds. What's wrong with this desktop app? Why it differs so much in performance comparing to Portal?

I see 624 storage accounts under the one subscription. Possibly there are which are not visible for me.

**Comment by MRayermannMSFT** [comment]
@Kotsiubynskyi 

> on Azure Portal the storage can be found within 2 seconds. What's wrong with this desktop app?

The Azure portal search experience uses a different API than Storage Explorer, at the trade off of being able to do/not do certain things. We looked at swapping to it in the past, but given the cost we decided to first try making it easier for users to filter down what they want to search in an attempt to make Storage Explorer tree view search faster by making it so Storage Explorer doesn't waste time searching for things you don't care about

> I see 624 storage accounts under the one subscription.

Just to clarify, you are saying that under the subscription you are searching you have only 624 storage accounts *total*, correct? If so, then the 3 minutes performance is bad. We need to fix that. That's so few storage accounts that we should be able to search all of them in effectively one API call. We will look into this further.
